internal/validations/validations.go

```diff
@@ -1,13 +1,30 @@
 package validations
 
 import (
+	"path/filepath"
 	"regexp"
 
 	"github.com/asaskevich/govalidator"
 	"github.com/aws/aws-sdk-go/aws/arn"
 	"github.com/distribution/distribution/reference"
 )
 
+var (
+	defaultAllowAll = []string{"**/*"}
+	imageURIs       = defaultAllowAll
+)
+
+// SetImageURIs restricts the approved container URIs to the provided set. To reset to a default allow-all state,
+// provide an empty list.
+func SetImageURIs(uris []string) {
+	if len(uris) == 0 {
+		imageURIs = defaultAllowAll
+		return
+	}
+
+	imageURIs = uris
```

Comment:
Given the change to globbing, indexed based lookups didn't make sense anymore. Reverted the set creation logic.